### DISCUSSION TITLE
Why Are Billing Fields Required for Virtual Products?

### DISCUSSION THREAD

**Discussion by braddalton** [woocommerce/woocommerce#49392] [Checkout Flow]
You have toggle fields to remove some billing fields but not others. Why are address fields required for virtual downloads?

**Comment by senadir** [comment]
An address is always required for an order, virtual orders still need to calculate taxes and pass fraud, and that's why addresses are needed there.

Check the comments here for more reasons and a code snippet to disable them

https://github.com/woocommerce/woocommerce/discussions/49617
